enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,287 @@
+valid_lft forever preferred_lft forever
+
+# worker 2
+3: ens4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
+link/ether 52:54:00:06:d6:58 brd ff:ff:ff:ff:ff:ff
+altname enp0s4
+inet 192.168.123.94/24 brd 192.168.123.255 scope global dynamic noprefixroute ens4
+valid_lft 3494sec preferred_lft 3494sec
+inet6 fe80::21df:c25a:dc63:f6eb/64 scope link noprefixroute
+valid_lft forever preferred_lft forever
+```
+
+Cluster manager would assign the `EgressIP` to worker 2.
+
+To allow packets to be steered into secondary networks, OVN reroute logical policies on `ovn_cluster_router` will redirect
+the packet out to the management port of the node which the egress IP is hosted.
+
+We will add rules in linux routing policy database control for each source Pod IP and lookup the correct routing table
+which will contain one route. This single route within a new managed routing table will direct the packet to the correct interface.
+
+We will also add a masquerade rule to iptables when a packet leaving a particular interface matches the source pod IP.
+
+
+See section `Design Details` for full implementation details.
+
+### User Stories
+As a user, I want the ability for cluster workload traffic to be routed to a network that is not managed by OVN and to have a configurable source IP.
+
+### Test Plan
+E2E tests verifying the correct source IP for deployment modes global IC and multi-zone IC:
+- Day 2 Networks CRUD including mask resizing
+- Configure new EgressIP both on ovn managed and unmanaged networks
+- Migrate existing EgressIP on ovn managed and unmanaged networks
+- Delete existing EgressIP on ovn managed and unmanaged networks
+- Add node
+- Delete node
+- Edit node
+- Add pod
+- Delete pod
+- Remove pod
+- Delete a valid rule manually 
+- Delete a valid IPtables rule manually
+- Delete a valid route manually
+- Delete a valid egress IP assigned to an interface manually
+- Valid egress service and egress IP functionality if both are active on the same k8 objs
+Today, if egress service and egress IP both 'match' on the same pod, egress service is preferred
+Continue this for this enhancement where egress service and egress IP overlap, prefer egress service
+- Egress IP interface is now a child interface and attached to a bond which is now the IP of the EIP interface. Reconcile to move the EIP to the bond.
+- Egress IP interface is now a child interface and attached to a bond which is now a different IP of the original EIP interface. Reconcile to move the EIP
+to a different network and if none found, clear EIP node status and let cluster manager reconcile and find a new node
+- Egress IP balancing of non-ovn EIP assignments is the same as current EIP balancing
+- Mixing Egress IPs from ovn and non-ovn managed networks
+- GARPs are sent when egress IP moves from a non-ovn managed network
+- Link goes down
+- EIP removed
+- Original network that allowed EIP assignment is removed
+
+### API Extensions
```

Comment:
so from what I understand there is no API change needed to indicate you want to use this new behavior, where we determine which interface suits the IP a user gave us. Is that right? I'm wondering is there an issue here, where a customer has been using egress IP and assuming it will go out their br-ex interface, and once they upgrade to this it could pick a different interface?

If this is a real issue, should we make this enabled via some new API?

Reply:
Can you see this actually happening in the field? 
Also, is that a problem if it goes out another interface? Theyll still get their traffic out into the right network but there will be a different MAC in the pkt than what they saw before.
If you think this is an issue, happy to extend the API.

Reply:
I could see it happening. Maybe we should use a detectNetwork (there is probably a better name you can come up with) field in the spec to determine whether this behavior is on or off?

Reply:
I dont think theres a need for a new field if we first try and always see if we can host the egress IP on the primary network and only after failing to host it there, we consider secondary interfaces.

Reply:
ok so by try you mean if the primary interface has a subnet that matches the requested egress IP, we prefer that one instead of doing longest prefix match?

Reply:
exactly